core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -152,12 +150,11 @@ public boolean isInSafeMode() {
   }
 
   @Override
-  @Nullable
   public InetSocketAddress getWebAddress() {
-    if (mWebServer != null) {
+    if (isWebServing()) {
```

Comment:
There could also be a worry for a null pointer exception on mWebServer on line 155, depending on where it is called. Also according to the interface this is overriding it should return null when not serving.

Reply:
I changed the interface: the callers never check for null.